pkg/services/publicdashboards/api/api.go

```diff
@@ -129,6 +133,18 @@ func (api *Api) ListPublicDashboards(c *models.ReqContext) response.Response {
 	return response.JSON(http.StatusOK, resp)
 }
 
+// Delete a public dashboard
+// DELETE /api/dashboards/:uid/public/:publicDashboardUid
+func (api *Api) DeletePublicDashboard(c *models.ReqContext) response.Response {
+	err := api.PublicDashboardService.Delete(c.Req.Context(), c.SignedInUser.OrgID, web.Params(c.Req)[":uid"], web.Params(c.Req)[":publicDashboardUid"])
+	if err != nil {
+		return api.handleError(c.Req.Context(), http.StatusInternalServerError, "DeletePublicDashboard: failed to delete public dashboard", err)
```

Comment:
In case of invalid request params we would be returning a 500. I think we should return bad request for those cases.